Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/dialogs/MotionAlertDialog.java

```diff
@@ -60,117 +51,100 @@ public boolean onKeyDown(int keyCode, KeyEvent event)
+	 *
+	 * @param keyEvent     KeyEvent of this key press.
+	 */
+	private void saveKeyInput(KeyEvent keyEvent)
 	{
-		return onKeyDown(event.getKeyCode(), event) || super.dispatchKeyEvent(event);
-	}
+		InputDevice device = keyEvent.getDevice();
+		String bindStr = "Device '" + device.getDescriptor() + "'-Button " + keyEvent.getKeyCode();
+		String uiString = device.getName() + ": Button " + keyEvent.getKeyCode();
 
-	@Override
-	public boolean dispatchGenericMotionEvent(MotionEvent event)
-	{
-		return onMotionEvent(event) || super.dispatchGenericMotionEvent(event);
+		saveInput(bindStr, uiString);
 	}
 
 	/**
-	 * Saves the provided input setting both to the INI file (so native code can use it) and as an
-	 * Android preference (so it persists correctly, and is human-readable.)
+	 * Saves the provided motion input setting both to the INI file (so native code can use it) and as
+	 * an Android preference (so it persists correctly and is human-readable.)
 	 *
-	 * @param device       Required; the InputDevice from which the input event originated.
-	 * @param keyEvent     If the event was a button push, this KeyEvent represents it and is required.
-	 * @param motionRange  If the event was an axis movement, this MotionRange represents it and is required.
-	 * @param axisPositive If the event was an axis movement, this boolean indicates the direction and is required.
+	 * @param device       InputDevice from which the input event originated.
+	 * @param motionRange  MotionRange of the movement
+	 * @param axisDir      Either '-' or '+'
 	 */
-	private void saveInput(InputDevice device, KeyEvent keyEvent, InputDevice.MotionRange motionRange, boolean axisPositive)
+	private void saveMotionInput(InputDevice device, InputDevice.MotionRange motionRange, char axisDir)
 	{
-		String bindStr = null;
-		String uiString = null;
+		String bindStr = "Device '" + device.getDescriptor() + "'-Axis " + motionRange.getAxis() + axisDir;
+		String uiString = device.getName() + ": Axis " + motionRange.getAxis() + axisDir;
 
-		if (keyEvent != null)
-		{
-			bindStr = "Device '" + device.getDescriptor() + "'-Button " + keyEvent.getKeyCode();
-			uiString = device.getName() + ": Button " + keyEvent.getKeyCode();
-		}
-
-		if (motionRange != null)
-		{
-			if (axisPositive)
-			{
-				bindStr = "Device '" + device.getDescriptor() + "'-Axis " + motionRange.getAxis() + "+";
-				uiString = device.getName() + ": Axis " + motionRange.getAxis() + "+";
-			}
-			else
-			{
-				bindStr = "Device '" + device.getDescriptor() + "'-Axis " + motionRange.getAxis() + "-";
-				uiString = device.getName() + ": Axis " + motionRange.getAxis() + "-";
-			}
-		}
-
-		if (bindStr != null)
-		{
-			setting.setValue(bindStr);
-		}
-		else
-		{
-			Log.error("[MotionAlertDialog] Failed to save input to INI.");
-		}
+		saveInput(bindStr, uiString);
+	}
 
+	/** Save the input string to settings and SharedPreferences, then dismiss this Dialog. */
```

Comment:
Shouldn't this also include `@param` instances for `bind` and `ui`?

Reply:
Not necessarily, I figured for a private method it would be suitable to leave the @params off. I can add them back, though, what do you think?

Reply:
Nah, it's fine. It's relatively self-explanatory in this case. Was more just a comment to ask if you had intended to put them there in the first place but had forgotten or something.